### ISSUE TITLE
[v3] Implement new Concept Exercise: string-formatting

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/fsharp#841]
This issue describes how to implement the `string-formatting` concept exercise for the F# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Integral Numbers in F#.

## Learning objectives

- Know how to use the `ToString()` method to convert any object to a `string`.
- Know how to use printf formatting on values of any type.
- Know what the benefit of printf formatting is over string concatenation.
- Know how to use default format strings to convert to standard output formats.
- Know how to use custom format strings to convert to custom output formats.
- Know how to combine strings using functions in the `String` module.
- Know of the `StringBuilder` type and when to use it.

## Out of scope

Nothing.

## Concepts

This Concepts Exercise's Concepts are:

- `string-formatting`: know how to use the `ToString()` method to convert any object to a `string`; know how to use printf formatting on values of any type; know what the benefit of printf formatting is over string concatenation; know how to use default format strings to convert to standard output formats; know how to use custom format strings to convert to custom output formats; know how to combine strings using functions in the `String` module; know of the `StringBuilder` type and when to use it.

## Prequisites

This Concept Exercise's prerequisites Concepts are:

- `strings`: strings will be formatted.
- `inheritance`: knowing that each class derives from `object` and thus has built-in methods.

Any data types used in this exercise (e.g. `datetimes`) should also be added as prerequisites.

### Hints

- [`String` module][string-module]: the `String` module.
- [Formatting text][formatting-text]: shows how to use the printf-family functions to format text.

The following links all use C# as their language, but are equally valid for F#:

- [Formatting types][formatting-types]: the available format types to use.
- [Standard numeric format strings][standard-numeric-format-strings]: lists the standard numeric format strings.
- [Custom numeric format strings][custom-numeric-format-strings]: describes how to define custom numeric format strings.
- [Standard date and time format strings][standard-date-and-time-format-strings]: lists the standard date and time format strings.
- [Custom date and time format strings][custom-date-and-time-format-strings]: describes how to define custom date and time format strings.
- [String builder][string-builder]: shows how to use the string builder.

### After

- [`String` module][string-module]: the `String` module.
- [Formatting text][formatting-text]: shows how to use the printf-family functions to format text.

The following links all use C# as their language, but are equally valid for F#:

- [Formatting types][formatting-types]: the available format types to use.
- [Standard numeric format strings][standard-numeric-format-strings]: lists the standard numeric format strings.
- [Custom numeric format strings][custom-numeric-format-strings]: describes how to define custom numeric format strings.
- [Standard date and time format strings][standard-date-and-time-format-strings]: lists the standard date and time format strings.
- [Custom date and time format strings][custom-date-and-time-format-strings]: describes how to define custom date and time format strings.
- [String builder][string-builder]: shows how to use the string builder.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/fsharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[how-to-implement-a-concept-exercise]: https://github.com/exercism/v3/blob/master/docs/maintainers/generic-how-to-implement-a-concept-exercise.md
[implemented-exercises]: https://github.com/exercism/v3/tree/master/languages/fsharp/exercises/concept/README.md#implemented-exercises
[reference]: https://github.com/exercism/v3/blob/master/reference/concepts/type_inference.md
[analyzer]: https://github.com/exercism/fsharp-analyzer
[representer]: https://github.com/exercism/fsharp-representer
[meta-design]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/design.md
[meta-config.json]: https://github.com/exercism/v3/blob/master/languages/fsharp/exercises/concept/numbers/.meta/config.json
[docs-concept-exercises]: https://github.com/exercism/v3/blob/master/docs/concept-exercises.md
[integral-number-types]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/integral-numeric-types
[string-builder]: https://docs.microsoft.com/en-us/dotnet/standard/base-types/stringbuilder
[string-module]: https://msdn.microsoft.com/visualfsharpdocs/conceptual/core.string-module-%5bfsharp%5d
[formatting-text]: https://fsharpforfunandprofit.com/posts/printf/
[formatting-types]: https://docs.microsoft.com/en-us/dotnet/standard/base-types/formatting-types
[standard-numeric-format-strings]: https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings
[custom-numeric-format-strings]: https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-numeric-format-strings
[standard-date-and-time-format-strings]: https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-date-and-time-format-strings
[custom-date-and-time-format-strings]: https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-date-and-time-format-strings
